### ISSUE TITLE
Build Error Linux ARM Chromebook Duet

### ISSUE THREAD

**Issue by sly-roccoon** [RedInquisitive/Super-Haxagon#18]
Hi,
I just tried compiling the project by downloading the repo, creating a build dir within the folder running 'cmake ..' and then 'make'.
I got this error:
```
/usr/bin/ld: CMakeFiles/SuperHaxagon.dir/source/States/Load.cpp.o: in function `SuperHaxagon::Load::enter()':
Load.cpp:(.text+0x1364): undefined reference to `std::filesystem::__cxx11::directory_iterator::operator*() const'
/usr/bin/ld: Load.cpp:(.text+0x14d8): undefined reference to `std::filesystem::__cxx11::directory_iterator::operator++()'
/usr/bin/ld: CMakeFiles/SuperHaxagon.dir/source/States/Load.cpp.o: in function `std::filesystem::__cxx11::path::path(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, std::filesystem::__cxx11::path::format)':
Load.cpp:(.text._ZNSt10filesystem7__cxx114pathC2EONSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEENS1_6formatE[_ZNSt10filesystem7__cxx114pathC5EONSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEENS1_6formatE]+0x4c): undefined reference to `std::filesystem::__cxx11::path::_M_split_cmpts()'
/usr/bin/ld: CMakeFiles/SuperHaxagon.dir/source/States/Load.cpp.o: in function `std::filesystem::__cxx11::operator==(std::filesystem::__cxx11::path const&, std::filesystem::__cxx11::path const&)':
Load.cpp:(.text._ZNSt10filesystem7__cxx11eqERKNS0_4pathES3_[_ZNSt10filesystem7__cxx11eqERKNS0_4pathES3_]+0x18): undefined reference to `std::filesystem::__cxx11::path::compare(std::filesystem::__cxx11::path const&) const'
/usr/bin/ld: CMakeFiles/SuperHaxagon.dir/source/States/Load.cpp.o: in function `std::filesystem::__cxx11::path::extension() const':
Load.cpp:(.text._ZNKSt10filesystem7__cxx114path9extensionEv[_ZNKSt10filesystem7__cxx114path9extensionEv]+0x18): undefined reference to `std::filesystem::__cxx11::path::_M_find_extension() const'
/usr/bin/ld: CMakeFiles/SuperHaxagon.dir/source/States/Load.cpp.o: in function `std::filesystem::__cxx11::directory_iterator::directory_iterator(std::filesystem::__cxx11::path const&)':
Load.cpp:(.text._ZNSt10filesystem7__cxx1118directory_iteratorC2ERKNS0_4pathE[_ZNSt10filesystem7__cxx1118directory_iteratorC5ERKNS0_4pathE]+0x20): undefined reference to `std::filesystem::__cxx11::directory_iterator::directory_iterator(std::filesystem::__cxx11::path const&, std::filesystem::directory_options, std::error_code*)'
/usr/bin/ld: CMakeFiles/SuperHaxagon.dir/source/States/Load.cpp.o: in function `std::filesystem::__cxx11::path::path<char [9], std::filesystem::__cxx11::path>(char const (&) [9], std::filesystem::__cxx11::path::format)':
Load.cpp:(.text._ZNSt10filesystem7__cxx114pathC2IA9_cS1_EERKT_NS1_6formatE[_ZNSt10filesystem7__cxx114pathC5IA9_cS1_EERKT_NS1_6formatE]+0x5c): undefined reference to `std::filesystem::__cxx11::path::_M_split_cmpts()'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/SuperHaxagon.dir/build.make:373: SuperHaxagon] Error 1
make[1]: *** [CMakeFiles/Makefile2:73: CMakeFiles/SuperHaxagon.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

Don't know what I did wrong :/

**Comment by RedTopper** [comment]
I'm going to take a shot in the dark here, but just going off the error alone, it looks like your compiler is wanting to use c++11 instead of c++17. I've only ever used the cmake files in an IDE like Visual Studio or CLion, so I'm not quite sure how to fix that. I was pretty sure the cmake file specified to use c++17 so this might not be the actual issue.

Could you see if you can get it to use the c++17 standard or maybe try out the traditional Makefile? I didn't consider ARM when programming it but I don't see why it shouldn't work.

If there needs to be actual code changes to support ARM I'd be happy to make them too!

**Comment by sly-roccoon** [comment]
The error happened after the make command, cmake worked fine. Should I check if make works correctly or cmake? I've never really gotten into make files in the first place so I'll have to look into that. ^^

**Comment by RedTopper** [comment]
Yeah, cmake will actually make a Makefile for you, as you just showed. I think it's more likely that you will get the cmake generated Makefile working over my Makefile that's in the root directory since I only wrote my Makefile with x86-64 in mind. But it's possible that running `make TARGET:=LINUX64` will work anyway (even though the target architecture is wrong)

With that said, maybe doing something like this might work for you:
https://stackoverflow.com/a/44477264

Basically add
`stdc++fs`
to the file
https://github.com/RedTopper/Super-Haxagon/blob/master/CMakeLists.txt#L67

**Comment by sly-roccoon** [comment]
Ok, so I added `set(CMAKE_CXX_FLAGS "-std=c++17 -lstdc++fs" )` now... (that's what you meant right? ^^') but it still spits out cxx11 stuff in the error output. So I think it's the same problem still...

**Comment by RedTopper** [comment]
Ha, sorry, that was a bit unclear.

Try editing the line to look like
`target_link_libraries(SuperHaxagon sfml-graphics sfml-window sfml-audio sfml-system stdc++fs)`

**Comment by sly-roccoon** [comment]
Ah well, no worries, I'm just a bit new to this.... Yeah this compiled it, but.... now I got a new error... 'Failed to open X11 display'... (Maybe should've mentioned I'm on a chromebook...)

But yeah the compiling issue was resolved now :D

**Comment by RedTopper** [comment]
Glad the issue was fixed!

Unfortunately I don't have a chromebook on hand to test the 'Failed to open X11 display' issue, so I can't say much about that.

However, if you manage to get it working then please feel free to make a PR for it. I'd be happy to see this game working on more hardware!
